### PR TITLE
bugfix/transcript-utils-retrieval

### LINKED ISSUE #95: Fix the `research_utils.transcripts.get_most_recent_transcript_manifest` function to actually pull all transcripts
Currently there is a bug when using the `CDPInstance` object because it is using the read only access database configuration that without a limit it can miss some of the transcripts because of how many items are in the file table.

The changes should do a couple of things:
1. After getting the list of events, use a ThreadPoolExecutor to pull the most recent transcript for each event.
2. After getting the transcripts, use a ThreadPoolExecutor to pull each transcripts file details.
3. Merge them all into the same dataframe.

### PR DESCRIPTION
**Pull request by JacksonMaxfield**
**Pull request recommendations:**
- [x] Name your pull request _your-development-type/short-description_. Ex: _feature/read-tiff-files_
- [x] Link to any relevant issue in the PR description. Ex: _Resolves [gh-12], adds tiff file format support_

Resolves #95 

- [x] Provide context of changes.

The bulk of these changes are entirely related to the transcript utils functions not being optimized for thousands of transcripts, files, and events. This will now start a threaded operation to get all the transcript information it needs instead of attempting to pull everything at once (that usually fails anyway due to the limits put on the the open access database queries).

Additionally, this adds functionality to choose whether to select the highest confidence transcripts or the most recent transcripts.

There are some other minor changes which include:
1. Finally fixing the `tox.ini` file to actually install all the dependencies required for the tests.
2. Re-configuring the test data and mocks to point at a fake CDP instance rather than at Seattle staging because it is much safer for testing.

- [x] Provide relevant tests for your feature or bug fix.
- [x] Provide or update documentation for any feature added by your pull request.

Thanks for contributing!

### CODE REVIEW

File: cdptools/research_utils/transcripts.py [hunk 1]
```diff
@@ -1,72 +1,154 @@
...
+    # Return none to be filtered out
+    return TrancriptJoin(event_id=event_id, transcript_details=None)
 
-def get_most_recent_transcript_manifest(db: Database) -> pd.DataFrame:
+
+def _get_file(file_id: str, db: Database) -> Dict[str, Any]:
+    return db.select_row_by_id(table="file", id=file_id)
+
+
+def get_transcript_manifest(db: Database, order_by_field: str = "confidence") -> pd.DataFrame:
```
Comment (isaacna) [inline cdptools/research_utils/transcripts.py]: This is great, I didn't know much about dataframes before but they are pretty cool and I learned a lot from this!

File: cdptools/research_utils/transcripts.py [hunk 2]
```diff
@@ -1,72 +1,154 @@
...
+            f"Provided `order_by_field` value is not a valid selection for transcript ordering / selection. "
+            f"Received: {order_by_field}. Possible choices: {ALLOWED_ORDER_BY_FIELDS}"
+        )
+
+    # Create transcript get partial
+    transcript_get = partial(_get_transcript_for_event, db=db, order_by_field=order_by_field)
+
+    # Threaded request all transcripts
+    with ThreadPoolExecutor() as exe:
+        transcript_joins = list(exe.map(transcript_get, list(events.event_id)))
```
Comment (isaacna) [inline cdptools/research_utils/transcripts.py]: I like the creation of the TranscriptJoin class that simplifies the code and allows ThreadPoolExecutor use, I probably would've tried an overcomplicated db query.

File: tox.ini [hunk 3]
```diff
@@ -19,6 +19,6 @@ commands =
 setenv =
     PYTHONPATH = {toxinidir}
 deps =
-    .[test]
+    .[all]
```
Comment (isaacna) [inline tox.ini]: Thanks for finding the bug! I made this change locally and it fixes `make build` for me

File: examples/transcripts.ipynb [hunk 4]
```diff
@@ -736,7 +736,7 @@
     }
    ],
    "source": [
-    "event_corpus_map = seattle.download_most_recent_transcripts()\n",
+    "event_corpus_map = seattle.download_event_transcripts()\n",
```
Comment (tohuynh) [inline examples/transcripts.ipynb]: Should this be `download_transcripts()` ?
Or change cdp_instance.py to have `download_event_transcripts()` instead

File: cdptools/research_utils/transcripts.py [hunk 5]
```diff
@@ -78,7 +160,7 @@ def download_most_recent_transcripts(
     save_dir.mkdir(parents=True, exist_ok=True)
 
     # Get most recent transcript data
-    most_recent = get_most_recent_transcript_manifest(db)
+    most_recent = get_transcript_manifest(db=db, order_by_field=order_by_field)
 
     # Begin storage
     most_recent.apply(
```
Comment (tohuynh) [inline cdptools/research_utils/transcripts.py]: I know this is not a apart of the changes, but is it possible to do a threaded download file here too?

**Comment by tohuynh** [comment]
I'm liking this. I tried running the example transcripts.ipynb and it works!

File: examples/transcripts.ipynb (hunk 4 above)
Comment (JacksonMaxfield) [reply examples/transcripts.ipynb]: Hmm I think you are right. There is no point to name it `download_event_transcripts` when we only have `event` transcripts to begin with. Will change and make downstream changes as well.

File: cdptools/research_utils/transcripts.py (hunk 5 above)
Comment (JacksonMaxfield) [reply cdptools/research_utils/transcripts.py]: Huh. I really thought `dataframe.apply` was already threaded for some reason. Will _gladly_ make the change. :+1:

**Pull request merged**